daemon/emer-persistent-cache.c

```diff
@@ -459,7 +459,13 @@ update_boot_offset (EmerPersistentCache *self,
         return save_timing_metadata (self, &relative_time, &absolute_time, NULL,
                                      NULL, NULL, error);
 
-      return TRUE;
+      /*
+       * Always make sure a file exists with the boot offset to maintain the
+       * invariant that when this function returns TRUE, a keyfile exists on
+       * disk.
+       */
+      return save_timing_metadata (self, NULL, NULL, &priv->boot_offset,
```

Comment:
We can't write to disk here because this code path gets hit every time an event is recorded. In particular, Roddy's constraint for the metrics system is that we write no more than once an hour. About to hop into a 1:1, but I can investigate why the test was failing after that if you like.